extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java

```diff
@@ -493,42 +494,19 @@ private void notifyLightStatusListeners(final FullLight fullLight, final String
     }
 
     /**
-     * Because the State can produce NPEs on getColorMode() and getEffect(), at first we check for the common
-     * properties which are set for every light type. If they equal, we additionally try to check the colorMode. If we
-     * get an NPE,
-     * the light does not support color mode and the common properties equality is our result: true. Otherwise if no NPE
```

Comment:
Hm, you removed the JavaDoc as you also changed the logic for the method - but is this the right thing to do?

If I see it correctly, the method so far potentially returned true if one state had a color mode, while the other hadn't. This seems to be no longer the case now.

Reply:
Ok, browsing the code, it seems that it was mainly done that complicated because we had no chance to fix the `State` class when it was still in the external library - we are in a more comfortable situations these days :-)